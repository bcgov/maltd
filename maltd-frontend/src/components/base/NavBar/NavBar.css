.nav-div {
  background-color: #cccccc;
}

@media only screen and (max-width: 768px) {
  .nav-div {
    padding-left: 3em;
  }
}
