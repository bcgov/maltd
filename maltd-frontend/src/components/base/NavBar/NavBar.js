/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { Collapse, Navbar, NavbarBrand, Nav, NavbarText } from "reactstrap";
import PropTypes from "prop-types";
import "./NavBar.css";
import GeneralButton from "../GeneralButton/GeneralButton";

const generalButton = {
  type: "submit",
  color: "danger",
  disabled: false,
  label: "Logout"
};

export default function NavBar({ navBar: { isAuthed } }) {
  return (
    <div className="nav-div">
      <Navbar expand="md">
        <NavbarBrand>MALT</NavbarBrand>
        <Collapse isOpen={true} navbar>
          <Nav className="mr-auto" navbar>
            <NavbarText>Account and License Management Tool</NavbarText>
          </Nav>
          {isAuthed && (
            <GeneralButton
              generalButton={generalButton}
              onClick={() => alert("Are you sure you want to log out?")}
            />
          )}
        </Collapse>
      </Navbar>
    </div>
  );
}

NavBar.propTypes = {
  navBar: PropTypes.shape({
    isAuthed: PropTypes.bool.isRequired
  }).isRequired
};
