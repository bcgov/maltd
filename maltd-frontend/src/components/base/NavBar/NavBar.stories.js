/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import NavBar from "./NavBar";

const navBar = {
  isAuthed: false
};

storiesOf("NavBar", module)
  .add("unauthenticated", () => <NavBar navBar={navBar} />)
  .add("authenticated", () => (
    <NavBar navBar={{ ...navBar, isAuthed: true }} />
  ));
