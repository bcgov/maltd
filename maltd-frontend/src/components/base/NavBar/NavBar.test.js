import React from "react";
import renderer from "react-test-renderer";
import NavBar from "./NavBar";

describe("NavBar", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <NavBar
        navBar={{
          isAuthed: false
        }}
        onClick={() => jest.fn()}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
