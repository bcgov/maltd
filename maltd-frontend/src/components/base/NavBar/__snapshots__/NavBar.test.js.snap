// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders as expected 1`] = `
<div
  className="nav-div"
>
  <nav
    className="navbar navbar-expand-md"
  >
    <a
      className="navbar-brand"
    >
      MALT
    </a>
    <div
      className="collapse show navbar-collapse"
      style={Object {}}
    >
      <ul
        className="mr-auto navbar-nav"
      >
        <span
          className="navbar-text"
        >
          Account and License Management Tool
        </span>
      </ul>
    </div>
  </nav>
</div>
`;
